extensions/warp-ipfs/src/js_exports.rs

```diff
@@ -0,0 +1,180 @@
+use std::str::FromStr;
+
+use crate::WarpIpfs;
+use js_sys::Uint8Array;
+use warp::crypto::DID;
+use warp::multipass::{
+    identity::{self, Identity, IdentityProfile},
+    MultiPass,
+};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct MultiPassBox {
```

Comment:
The name may not be suitable here. Not sure what the proper name should be though.

Reply:
I wonder if it might be better to create a macro rule so we could pass `WarpIpfs` (or any struct or item that implements `MultiPass` and its traits) so we could do something like `generate_wam!(WarpIpfs);`  when the target is `wasm32-unknown-unknown` it would generate the functions accordingly, maybe even a constructor function that could be used from js?

Reply:
As discussed, a new `WarpIpfs` returns a `WarpInstance` which contains all the boxed traits.